addOns/frontendscanner/src/test/java/org/zaproxy/zap/extension/frontendscanner/FrontEndScannerProxyListenerUnitTest.java

```diff
@@ -0,0 +1,244 @@
+                        + HOSTNAME
+                        + "\\/zapCallBackUrl\\/-?[0-9]+\\?action=getFile&filename=front-end-scanner.js&historyReferenceId=42'><\\/script><\\/head><body><\\/body></html>";
+        String result = msg.getResponseBody().toString();
+
+        assertTrue(result.matches(expectedHtmlFormat));
+    }
+
+    @Test
+    public void testInjectAfterAllMetaTagsInHeadTag() {
+        // Given
+        when(options.isEnabled()).thenReturn(true);
+        String htmlBody =
+                "<!doctype html><html lang='en'><head><meta><meta></head><body></body></html>";
+        msg.setResponseBody(htmlBody);
+
+        // When
+        frontEndScannerProxyListener.onHttpResponseReceive(msg);
+
+        // Then
+        String expectedHtmlFormat =
+                "<!doctype html><html lang='en'><head><meta><meta><script src='https:\\/\\/"
+                        + HOSTNAME
+                        + "\\/zapCallBackUrl\\/-?[0-9]+\\?action=getFile&filename=front-end-scanner.js&historyReferenceId=42'><\\/script><\\/head><body><\\/body></html>";
+        String result = msg.getResponseBody().toString();
+
+        assertTrue(result.matches(expectedHtmlFormat));
+    }
+
+    @Test
+    public void testInjectionShouldBeSuccessfulWithoutHead() {
+        // Given
+        when(options.isEnabled()).thenReturn(true);
+        String htmlBody = "<!doctype html><html lang='en'><body></body></head>";
+        msg.setResponseBody(htmlBody);
+
+        // When
+        frontEndScannerProxyListener.onHttpResponseReceive(msg);
+
+        // Then
+        String expectedHtmlFormat =
+                "<!doctype html><html lang='en'><head><script src='https:\\/\\/"
+                        + HOSTNAME
+                        + "\\/zapCallBackUrl\\/-?[0-9]+\\?action=getFile&filename=front-end-scanner.js&historyReferenceId=42'><\\/script><\\/head><body><\\/body></head>";
+        String result = msg.getResponseBody().toString();
+
+        assertTrue(result.matches(expectedHtmlFormat));
+    }
+
+    @Test
+    public void testInjectionShouldBeSuccessfulWithEmptyHead() {
+        // Given
+        when(options.isEnabled()).thenReturn(true);
+        String htmlBody = "<!doctype html><html lang='en'><head></head><body></body></html>";
+        msg.setResponseBody(htmlBody);
+
+        // When
+        frontEndScannerProxyListener.onHttpResponseReceive(msg);
+
+        // Then
+        String expectedHtmlFormat =
+                "<!doctype html><html lang='en'><head><script src='https:\\/\\/"
+                        + HOSTNAME
+                        + "\\/zapCallBackUrl\\/-?[0-9]+\\?action=getFile&filename=front-end-scanner.js&historyReferenceId=42'><\\/script><\\/head><body><\\/body></html>";
+        String result = msg.getResponseBody().toString();
+
+        assertTrue(result.matches(expectedHtmlFormat));
+    }
+
+    @Test
+    public void testInjectionShouldBeSuccessfulWithHeadNotAtTheStartOfHtml() {
```

Comment:
This one is now somewhat redundant with the other tests, now that all are using the head not at the start. Would be great to restore one of the previous tests that had just head to keep covering that part.